src/randoop/operation/ArrayCreation.java

```diff
@@ -54,30 +67,41 @@ public ArrayCreation(Class<?> elementType, int length) {
     // Set state variables.
     this.elementType = elementType;
     this.length = length;
+    this.outputTypeCached = Array.newInstance(elementType, 0).getClass();
   }
 
+  /**
```

Comment:
I'm wondering why this class defines writeReplace (along with another class that defines readResolve) rather than this class just defining writeObject and readObject, which would keep all the code together in this file.